tests/time_tests/run_executable.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""
+This script runs TimeTests executable several times to aggregate
+collected statistics.
+"""
+
+import statistics as Statistics
+from pathlib import Path
+import tempfile
+import subprocess
+import logging
+import argparse
+import sys
+import yaml
+
+
+def run_cmd(args: list, log=None, verbose=True):
+    """ Run command
+    """
+    if log is None:
+        log = logging.getLogger('run_cmd')
+    log_out = log.info if verbose else log.debug
+
+    log.info(f'========== cmd: {" ".join(args)}')  # pylint: disable=logging-format-interpolation
+
+    proc = subprocess.Popen(args,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT,
+                            encoding='utf-8',
+                            universal_newlines=True)
+    output = []
+    for line in iter(proc.stdout.readline, ''):
+        log_out(line.strip('\n'))
+        output.append(line)
+        if line or proc.poll() is None:
+            continue
+        break
+    outs = proc.communicate()[0]
+
+    if outs:
+        log_out(outs.strip('\n'))
+        output.append(outs)
+    log.info('========== Completed. Exit code: %d', proc.returncode)
+    return proc.returncode, ''.join(output)
+
+
+def read_statistics(statistics_path, statistics: dir):
+    """Read statistics from a file and extend provided statistics"""
+    with open(statistics_path, "r") as file:
+        parsed_data = yaml.load(file, Loader=yaml.FullLoader)
+    return dict((step_name, statistics.get(step_name, []) + [duration])
+                for step_name, duration in parsed_data.items())
+
+
+def aggregate_statistics(statistics: dir):
+    """Aggregate provided statistics"""
+    return {step_name: {"avg": Statistics.mean(duration_list),
+                        "stdev": Statistics.stdev(duration_list)}
+            for step_name, duration_list in statistics.items()}
+
+
+def write_aggregated_stats(statistics_path, statistics: dir):
+    """Write aggregated statistics to a file in YAML format"""
+    with open(statistics_path, "w") as file:
+        yaml.dump(statistics, file)
+
+
+def run_executable(executable: Path, model: Path, device, niter, log=None):
```

Comment:
Should we make this function independent from the arguments of the executable? This way we have more flexibility supporting for example banchmar_app and also .

```
run_niter(args, niter, statistics_path, log=.., verbose=..):
 ...

mkdirs(os.path.split(statistics_path)[0], exists_ok=True)
run_niter([time_test, f'-s={statistics_path}'], 10, statistics_path)
```

Reply:
Good idea. Anyway, we have some difference in CLI (e.g. `-s` in `TimeTests`), so we can't pass arguments as is

Need to investigate it more

Reply:
Added `prepare_executable_cmd` that should provide flexibility for it